### PR TITLE
Add Infracost

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* [x] Add Infracost label
* [x] allow setting global labels across all stacks (escape hatch for the next label similar to infracost)
* [x] Pull the infracost api key from ssm and feed it into each stack to prevent a hard coded value in `.spacelift/config.yaml`.
    * [x] add input for the token (supplied by ssm)
    * [x] create global context
    * [x] use token to create env var and assoc to context 
    * [x] assign each stack the global context
* [x] test and verify

## why
* Get infracost per stack

## references
* https://docs.spacelift.io/vendors/terraform/infracost

Once this is enabled, you would then need to add the env var in the `.spacelift/config.yaml` file or the stack global context

```
INFRACOST_API_KEY
```

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -37,6 +39,9 @@ locals {
   distinct_policy_names = distinct(compact(flatten([
     for k, v in local.spacelift_stacks : try(v.settings.spacelift.policies_by_name_enabled, var.policies_by_name_enabled) if v.enabled
   ])))
+
+  # Concatenate labels with infracost if it's enabled
+  labels = var.infracost_enabled ? concat(var.labels, ["infracost"]) : var.labels
```
Comment (aknysh) [inline main.tf:44]: `var.labels` is not a variable for `main.tf`, it's in https://github.com/cloudposse/terraform-spacelift-cloud-infrastructure-automation/blob/master/modules/stack/variables.tf#L104

**Review by aknysh — COMMENTED** [review]
please see comments

File: main.tf (hunk 1 above)
Comment (nitrocode) [reply main.tf:44]: It is a variable in `main.tf` in this PR, I added it to `variables.tf` ?

Do you mean it shouldn't be an input for this module ?

Comment (aknysh) [reply main.tf:44]: oh sorry, did not notice the new variable

**Review by aknysh — APPROVED** [review]

**Pull request merged**